pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,641 @@
+
+	workerNode = &st.MakeNode().Name("worker").Label("nodename", "worker").Node
+
+	claim = st.MakeResourceClaim().
+		Name(claimName).
+		Namespace(namespace).
+		ResourceClassName(className).
+		Obj()
+	pendingImmediateClaim = st.FromResourceClaim(claim).
+				AllocationMode(resourcev1alpha1.AllocationModeImmediate).
+				Obj()
+	pendingDelayedClaim = st.FromResourceClaim(claim).
+				AllocationMode(resourcev1alpha1.AllocationModeWaitForFirstConsumer).
+				Obj()
+	pendingDelayedClaim2 = st.FromResourceClaim(pendingDelayedClaim).
+				Name(claimName2).
+				Obj()
+	deallocatingClaim = st.FromResourceClaim(pendingImmediateClaim).
+				Allocation(&resourcev1alpha1.AllocationResult{}).
+				DeallocationRequested(true).
+				Obj()
+	inUseClaim = st.FromResourceClaim(pendingImmediateClaim).
+			Allocation(&resourcev1alpha1.AllocationResult{}).
+			ReservedFor(resourcev1alpha1.ResourceClaimConsumerReference{UID: types.UID(podUID)}).
+			Obj()
+	allocatedClaim = st.FromResourceClaim(pendingDelayedClaim).
+			OwnerReference(podName, podUID, podKind).
+			Allocation(&resourcev1alpha1.AllocationResult{}).
+			Obj()
+	allocatedClaimWithWrongTopology = st.FromResourceClaim(allocatedClaim).
+					Allocation(&resourcev1alpha1.AllocationResult{AvailableOnNodes: st.MakeNodeSelector().In("no-such-label", []string{"no-such-value"}).Obj()}).
+					Obj()
+	allocatedClaimWithGoodTopology = st.FromResourceClaim(allocatedClaim).
+					Allocation(&resourcev1alpha1.AllocationResult{AvailableOnNodes: st.MakeNodeSelector().In("nodename", []string{"worker"}).Obj()}).
+					Obj()
+	otherClaim = st.MakeResourceClaim().
+			Name("not-my-claim").
+			Namespace(namespace).
+			ResourceClassName(className).
+			Obj()
+
+	scheduling = st.MakePodScheduling().Name(podName).Namespace(namespace).
+			OwnerReference(podName, podUID, podKind).
+			Obj()
+	schedulingPotential = st.FromPodScheduling(scheduling).
+				PotentialNodes(workerNode.Name).
+				Obj()
+	schedulingSelectedPotential = st.FromPodScheduling(schedulingPotential).
+					SelectedNode(workerNode.Name).
+					Obj()
+	schedulingInfo = st.FromPodScheduling(schedulingPotential).
+			ResourceClaims(resourcev1alpha1.ResourceClaimSchedulingStatus{Name: resourceName},
+			resourcev1alpha1.ResourceClaimSchedulingStatus{Name: resourceName2}).
+		Obj()
+)
+
+// result defines the expected outcome of some operation. It covers
+// operation's status and the state of the world (= objects).
+type result struct {
+	status *framework.Status
+
+	// changes contains a mapping of name to an update function for
+	// the corresponding object. These functions apply exactly the expected
+	// changes to a copy of the object as it existed before the operation.
+	changes changeMapping
+
+	// added contains objects created by the operation.
+	added []metav1.Object
+
+	// removed contains objects deleted by the operation.
+	removed []metav1.Object
+}
+
+type resultPreFilter struct {
+	result
+	preFilterResult *framework.PreFilterResult
+}
```

Comment:
Storing `preFilterResult` inside a special struct is unnecessarily complex. It becomes simpler when stored directly in `want` - will change.

Reply:
Done.